I tried using travis with my test added on top of the unfixed version. Some builds fail because of linting, however they all seem to pass pytest. I realised that my earlier version might have passed since `_IMPORTS` is set to true after the first call to `read_html()`. So I imported `pandas.io.html` and reloaded it. But this didn't fix the issue. This change was included in the version that was tested by travis.

These are the additions that were made to `test_html.py`:
```python
try:
    from importlib import reload
except ImportError:
    pass
[...]
import pandas.io.html
[...]
class ErrorThread(threading.Thread):
    def run(self):
        try:
            super(ErrorThread, self).run()
        except Exception as e:
            self.err = e
        else:
            self.err = None

@pytest.mark.slow
def test_importcheck_thread_safety():
    reload(pandas.io.html)
    filename = os.path.join(DATA_PATH, 'valid_markup.html')
    helper_thread1 = ErrorThread(target = read_html, args = (filename,))
    helper_thread2 = ErrorThread(target = read_html, args = (filename,))
    helper_thread1.start()
    helper_thread2.start()
    while(helper_thread1.is_alive() or helper_thread2.is_alive()):
        pass
    assert None == helper_thread1.err == helper_thread2.err
```
Output of `pytest test_html.py::test_importcheck_thread_safety`:
```
=========================== test session starts ============================
platform linux -- Python 3.6.1, pytest-3.1.1, py-1.4.34, pluggy-0.4.0
rootdir: /home/fe/code/pandas, inifile: setup.cfg
plugins: xdist-1.16.0
collected 8 items 

test_html.py F

================================= FAILURES =================================
______________________ test_importcheck_thread_safety ______________________

    @pytest.mark.slow
    def test_importcheck_thread_safety():
        reload(pandas.io.html)
        filename = os.path.join(DATA_PATH, 'valid_markup.html')
        helper_thread1 = ErrorThread(target = read_html, args = (filename,))
        helper_thread2 = ErrorThread(target = read_html, args = (filename,))
        helper_thread1.start()
        helper_thread2.start()
        while(helper_thread1.is_alive() or helper_thread2.is_alive()):
            pass
>       assert None == helper_thread1.err == helper_thread2.err
E       AssertionError: assert None == ImportError('lxml not found, please install it',)
E        +  where None = <ErrorThread(Thread-1, stopped 140146488710912)>.err
E        +  and   ImportError('lxml not found, please install it',) = <ErrorThread(Thread-2, stopped 140146480318208)>.err

test_html.py:961: AssertionError
========================= 1 failed in 7.14 seconds =========================
```
Output of `pytest test_html.py`
```
=========================== test session starts ============================
platform linux -- Python 3.6.1, pytest-3.1.1, py-1.4.34, pluggy-0.4.0
rootdir: /home/fe/code/pandas, inifile: setup.cfg
plugins: xdist-1.16.0
collected 74 items 

test_html.py ..........................................................................

======================== 74 passed in 50.77 seconds ========================
```

--- a/doc/source/whatsnew/v0.21.0.txt
+++ b/doc/source/whatsnew/v0.21.0.txt
@@ -164,6 +164,8 @@ I/O
 
 - Bug in :func:`read_stata` where value labels could not be read when using an iterator (:issue:`16923`)
 
+- Bug in :func:`read_html` where import check fails when run in multiple threads (:issue:`16928`)
+
 Plotting
 ^^^^^^^^
 - Bug in plotting methods using ``secondary_y`` and ``fontsize`` not setting secondary axis font size (:issue:`12565`)

--- a/pandas/io/html.py
+++ b/pandas/io/html.py
@@ -37,8 +37,6 @@ def _importers():
     if _IMPORTS:
         return
 
-    _IMPORTS = True
-
     global _HAS_BS4, _HAS_LXML, _HAS_HTML5LIB
 
     try:
@@ -58,6 +56,8 @@ def _importers():
         _HAS_HTML5LIB = True
     except ImportError:
         pass
+
+    _IMPORTS = True
 
 
 #############

--- a/pandas/tests/io/test_html.py
+++ b/pandas/tests/io/test_html.py
@@ -3,12 +3,16 @@ from __future__ import print_function
 import glob
 import os
 import re
+import threading
 import warnings
 
+
+# imports needed for Python 3.x but will fail under Python 2.x
 try:
-    from importlib import import_module
+    from importlib import import_module, reload
 except ImportError:
     import_module = __import__
+
 
 from distutils.version import LooseVersion
 
@@ -22,6 +26,7 @@ from pandas import (DataFrame, MultiIndex, read_csv, Timestamp, Index,
 from pandas.compat import (map, zip, StringIO, string_types, BytesIO,
                            is_platform_windows, PY3)
 from pandas.io.common import URLError, urlopen, file_path_to_url
+import pandas.io.html
 from pandas.io.html import read_html
 from pandas._libs.parsers import ParserError
 
@@ -931,3 +936,32 @@ def test_same_ordering():
     dfs_lxml = read_html(filename, index_col=0, flavor=['lxml'])
     dfs_bs4 = read_html(filename, index_col=0, flavor=['bs4'])
     assert_framelist_equal(dfs_lxml, dfs_bs4)
+
+
+class ErrorThread(threading.Thread):
+    def run(self):
+        try:
+            super(ErrorThread, self).run()
+        except Exception as e:
+            self.err = e
+        else:
+            self.err = None
+
+
+@pytest.mark.slow
+def test_importcheck_thread_safety():
+    # see gh-16928
+
+    # force import check by reinitalising global vars in html.py
+    reload(pandas.io.html)
+
+    filename = os.path.join(DATA_PATH, 'valid_markup.html')
+    helper_thread1 = ErrorThread(target=read_html, args=(filename,))
+    helper_thread2 = ErrorThread(target=read_html, args=(filename,))
+
+    helper_thread1.start()
+    helper_thread2.start()
+
+    while helper_thread1.is_alive() or helper_thread2.is_alive():
+        pass
+    assert None is helper_thread1.err is helper_thread2.err

